exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/text/TextFormatPlugin.java

```diff
@@ -343,4 +344,14 @@ protected ScanStats getScanStats(final PlannerSettings settings, final EasyGroup
     final double estRowCount = data / estimatedRowSize;
     return new ScanStats(GroupScanProperty.NO_EXACT_ROW_COUNT, (long) estRowCount, 1, data);
   }
+
+  @Override
+  public void setSchema(TupleMetadata schema) {
+    this.schema = schema;
+  }
```

Comment:
Would be best if the schema can be immutable: set in the constructor and not changeable once set. Else, it is very unclear what the semantics will be.

Same is true, by the way, for the scan nodes.